emcc

```diff
@@ -1085,6 +1081,15 @@ try:
     logging.debug('just preprocessor ' + ' '.join(cmd))
     exit(subprocess.call(cmd))
 
+  if '-M' in newargs or '-MM' in newargs:
```

Comment:
isn't this identical to the `-E` block of code right above it, except for the logged text? perhaps these can be merged (with just logging kept different)